Core/Util/PPGeDraw.cpp

```diff
@@ -955,12 +955,24 @@ static void PPGeDecimateTextImages(int age) {
 }
 
 void PPGeDrawText(const char *text, float x, float y, const PPGeStyle &style) {
-	if (!text || !strlen(text)) {
+	if (!text) {
+		return;
+	}
+	// Seen in Ratchet & Clank - Secret Agent. To match the output of the real thing, we have to remove
+	// both the overlong encoding and "ENTR", whatever that is. If we just let SanitizeUTF8 remove
+	// the overlong null, the rest of the string is missing in the bottom left corner (save size, etc).
+	// It doesn't seem to be using sceCcc.
+	// Note how the double "" is required in the middle of the string to end the \x80 constant (otherwise it takes E).
+	// TODO: Potentially if the string is only ended by a C080, ReplaceAll might overshoot :(
+	std::string str = ReplaceAll(text, "\xC0\x80""ENTR", "");
```

Comment:
Wouldn't `SanitizeUTF8` already stop before the ENTR anyway?  Is this needed?

I want to do some tests and see how other bad encodings perform, but have been really busy recently...

-[Unknown]

Reply:
It would if we ran it first.